Competitive Coding/Graphs/unionfind/CycleDetection.java

```diff
@@ -0,0 +1,66 @@
+import java.util.*;
+import java.lang.*;
+import java.io.*;
+class Graph
+{   int V, E;
+    Edge edge[];
+    class Edge
+    {
+        int src, dest;
+    };
+    Graph(int v,int e)
+    {
+        V = v;
+        E = e;
+        edge = new Edge[E];
+        for (int i=0; i<e; ++i)
+            edge[i] = new Edge();
+    }
+    int find(int parent[], int i)
+    {
+        if (parent[i] == -1)
+            return i;
+        return find(parent, parent[i]);
+    }
+    void Union(int parent[], int x, int y)
+    {
+        int xset = find(parent, x);
```

Comment:
document the code
and please show me the output

Reply:
do you want me to show the screen shot of the poutput?

Reply:
do you want me to show the screen shot of the output?

Reply:
some gif if possible

Reply:
![image](https://user-images.githubusercontent.com/30727381/33664567-1974984c-daba-11e7-8fee-4924293f273e.png)

Reply:
that's not much understandable

Reply:
I put everything in the code and made the output look simpler . I can add the comments in the code if you wish and maybe then it will look be more understandable .